src/dbnode/encoding/proto/encoder.go

```diff
@@ -298,14 +298,14 @@ func (enc *Encoder) LastEncoded() (ts.Datapoint, error) {
 	return enc.lastEncodedDP, nil
 }
 
-// LastAnnotation returns the last encoded annotation (which contain the bytes
+// LastAnnotation returns the checksum of the last encoded annotation (which contain the bytes
 // used for ProtoBuf data).
-func (enc *Encoder) LastAnnotation() (ts.Annotation, error) {
+func (enc *Encoder) LastAnnotationChecksum() (uint64, error) {
 	if enc.numEncoded == 0 {
-		return nil, errNoEncodedDatapoints
+		return 0, errNoEncodedDatapoints
 	}
 
-	return enc.prevAnnotation, nil
+	return xxhash.Sum64(enc.prevAnnotation), nil
```

Comment:
Better to convert `prevAnnotation` [here](https://github.com/m3db/m3/blob/master/src/dbnode/encoding/proto/encoder.go#L172) to be a `prevAnnotationChecksum` similarly to how its done in the timestamp encoder?

Reply:
Hard to say. This encoder is somewhat different - it does not have this elimination of repeated values, so it is not using `prevAnnotation` for its own needs, so I did not want to make it compute the checksum which would probably be not needed in many cases (looks like it is only needed when using this encoder from `dbBuffer`).
@robskillington what would you say?